repos/spack_repo/builtin/packages/hipblaslt/package.py

```diff
@@ -180,13 +200,39 @@ def patch(self):
                 string=True,
             )
         if not self.spec["hip"].external:
-            if self.spec.satisfies("@6.4:") and self.run_tests:
+            if self.spec.satisfies("@6.4:7.0") and self.run_tests:
                 filter_file(
                     r"${HIP_CLANG_ROOT}/lib",
                     "{0}/lib".format(self.spec["rocm-openmp-extras"].prefix),
                     "clients/CMakeLists.txt",
                     string=True,
                 )
+            if self.spec.satisfies("@7.1") and self.run_tests:
+                filter_file(
+                    r"${HIP_CLANG_ROOT}/lib",
+                    "{0}/lib".format(self.spec["rocm-openmp-extras"].prefix),
+                    "projects/hipblaslt/clients/CMakeLists.txt",
+                    string=True,
+                )
+        if self.spec.satisfies("@7.1:"):
+            yaml_path = f"{self.spec['py-pyyaml'].prefix}/lib/python{py_ver}/site-packages"
+            packaging_path = f"{self.spec['py-packaging'].prefix}/lib/python{py_ver}/site-packages"
+            msgpack_path = f"{self.spec['py-msgpack'].prefix}/lib/python{py_ver}/site-packages"
```

Comment:
Constructing these paths yourself probably works on most systems but it's not guaranteed (and may not work with some pythons). You can ask the python package for `purelib` or `platlib`, though:

```suggestion
            purelib = self.spec["python"].package.purelib
            yaml_path = os.path.join(self.spec['py-pyyaml'].prefix, purelib)
            packaging_path = os.path.join(self.spec['py-packaging'].prefix, purelib)
            msgpack_path = os.path.join(self.spec['py-msgpack'].prefix, purelib)
```

This also avoids issues w/windows paths, and you don't actually have to get the python version (it's built into purelib for you). I am not sure if you want purelib or platlib -- look at the `package.py` for python to see what they do.

Reply:
thanks I've changed it to use `purelib`